File: packages/java/parser-jvm-plugin-backbone/src/test/java/dev/hilla/parser/plugins/backbone/transients/TransientEndpoint.java
package dev.hilla.parser.plugins.backbone.transients;

@Endpoint
public class TransientEndpoint {
    public TransientModel getTransientModel() {
        return new TransientModel();
    }

    public static class TransientModel {
        private NonTransientEntity nonTransientEntity;
        private String notTransientField;
        private transient TransientEntity transientEntity;
        private transient String transientField;
    }
}

File: packages/java/parser-jvm-plugin-backbone/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/transients/expected/TransientEndpoint.ts
import { EndpointRequestInit as EndpointRequestInit_1 } from "@vaadin/hilla-frontend";
import type TransientModel_1 from "./com/vaadin/hilla/parser/plugins/backbone/transients/TransientEndpoint/TransientModel.js";
import client_1 from "./connect-client.default.js";
async function getTransientModel_1(init?: EndpointRequestInit_1): Promise<TransientModel_1 | undefined> { return client_1.call("TransientEndpoint", "getTransientModel", {}, init); }
export { getTransientModel_1 as getTransientModel };
